pkg/api/errors/errors.go

```diff
@@ -297,6 +297,17 @@ func NewGenericServerResponse(code int, verb, kind, name, serverMessage string,
 	case len(kind) > 0:
 		message = fmt.Sprintf("%s (%s %s)", message, strings.ToLower(verb), kind)
 	}
+	var causes []api.StatusCause
+	if isUnexpectedResponse {
+		causes = []api.StatusCause{
+			{
+				Type:    api.CauseTypeUnexpectedServerResponse,
+				Message: serverMessage,
+			},
+		}
+	} else {
+		causes = nil
```

Comment:
go-restful will return 404 when the path isn't registered? We could create a cause for that, to distinguish it from a proper path but no object.

go-restful will return 405 when the path is registered, but the verb is not? That error is perhaps specific enough without a cause.